azkaban-common/src/main/java/azkaban/project/ProjectManager.java

```diff
@@ -312,7 +312,13 @@ public Props getProperties(final Project project, final Flow flow, final String
       final String source) throws ProjectManagerException {
     if (FlowLoaderUtils.isAzkabanFlowVersion20(flow.getAzkabanFlowVersion())) {
       // Return the properties from the original uploaded flow file.
-      return getPropertiesFromFlowFile(flow, jobName, source, 1);
+      if (jobName != null) {
+        return getPropertiesFromFlowFile(flow, jobName, source, 1);
+      } else {
+        // If job is null we intend to fetch the top level flow properties, and flowID does not
+        // have the .flow extension suffix.
+        return getPropertiesFromFlowFile(flow, null, source + Constants.FLOW_FILE_SUFFIX, 1);
+      }
```

Comment:
I tested this change, it breaks fetching the flow properties page for flow 2.0 dags (we get to that page by clicking in the link under `Inherited Properties` section in the screenshot attached.)

To cover the cases of flow 2.0 in `ContainerImplUtils.getProxyUsersForFlow()` could you try something like this?
```
String flowFileName = null;
if(FlowLoaderUtils.isAzkabanFlowVersion20(flow.getAzkabanFlowVersion())) {
  flowFileName = flow.getFlowId() + Constants.FLOW_FILE_SUFFIX;
}
....
Props flowProps = projectManager.getProperties(project, flowObj, flowFileName, flow.getFlowId());
```
Note that this works IF the flow id matches the name of the dag file **IF there is no such requirement/validation then there is a chance that this code won't work either**.

<img width="1350" alt="Screenshot 2023-03-05 at 1 05 43 AM" src="https://user-images.githubusercontent.com/44478711/222952262-8c4cacd8-e3d8-403b-9390-6a0c9ff4d650.png">

Reply:
reverted.